components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -680,4 +687,17 @@ void ESP8266Interface::proc_oob_evnt()
         _esp.bg_process_oob(ESP8266_RECV_TIMEOUT, true);
     }
 }
+
+#define ESP8266_OOB_BUSY_TIMEOUT_MS 100
+
+void ESP8266Interface::_oob_busy_timeout()
+{
+    _esp.oob_busy_timeout();
+}
+
+void ESP8266Interface::_oob_busy_detected()
+{
+    _global_event_queue->call_in(ESP8266_OOB_BUSY_TIMEOUT_MS, callback(this, &ESP8266Interface::_oob_busy_timeout));
```

Comment:
An event should be added to the queue only if we are certain that "busy s..." or "busy p..." has occurred.

Reply:
_oob_busy_detected is called from ESP8266::_oob_busy via a callback. 
In the _oob_busy callback I can see only three way - "busy s...", "busy... p" or MBED_ERROR which would halt the system.
So I think the requirement is fulfilled?